edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.Http/controllers/IHttpRequestAuthenticator.cs

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Hub.Http.Controllers
+{
+    using System.Threading.Tasks;
+    using Microsoft.AspNetCore.Http;
+    using Microsoft.Azure.Devices.Edge.Hub.Core.Identity;
+    using Microsoft.Azure.Devices.Edge.Util;
+
+    public struct HttpAuthResult
+    {
+        public HttpAuthResult(bool authenticated, string errorMsg)
+        {
+            this.Authenticated = authenticated;
+            this.ErrorMsg = Preconditions.CheckNotNull(errorMsg);
+        }
+
+        public bool Authenticated { get; }
+        public string ErrorMsg { get; }
+    }
+
+    public interface IHttpRequestAuthenticator
+    {
+        Task<HttpAuthResult> AuthenticateRequest(string deviceId, Option<string> moduleId, HttpContext context);
```

Comment:
You can return a Try<bool> instead

Reply:
I was trying to avoid using exceptions to wrap the error message, seems like exceptions should be reserved for when something goes wrong as opposed to expected auth errors.